operations/firmware/Updating_Firmware_m001.md

```diff
@@ -0,0 +1,85 @@
+## Updating Firmware / BIOS for NCN m001
+
+**Run on m001**
+
+**Prerequisite:**
+* IP Address of m001 bmc
+* IP Address of m001
+* Root password for m001 bmc
+
+### Find the Model Name
+
+***On HPE Nodes***
+
+`m001# curl -k -u root:password https://ipaddressOfBMC/redfish/v1/Systems/1 | jq .Model`
+
+***On Gigbyte Nodes***
+
+`m001# curl -k -u root:password https://ipaddressOfBMC/redfish/v1/Systems/Self | jq .Model`
+
+### Get the Firmware Images
+
+`m001# cray fas images list --format json | jq .[] | .[] | select(.models | index("ModelName"))`
+where "ModelName" is the name from above
+
+This will give you a list of images stored inside FAS ready to be flashed.  Locate the image you need to flash m001 firmware and/or BIOS.
+
+Look at the s3URL for that command it will look like this:
+
+    "s3URL": "s3:/fw-update/4e5f569a603311eb96b582a8e219a16d/image.RBU"
+
+Using the s3URL path run the following command
+
+`m001# cray artifacts get fw-update 4e5f569a603311eb96b582a8e219a16d/image.RBU image.RBU`
+
+Where 4e5f569a603311eb96b582a8e219a16d/image.RBU is the path in the s3URL
+And image.RBU is the name of the file to save the image on local disk
+
+## Flash the Firmware
+
+***For Gigabyte m001***

```

Comment:
**Gigabyte ncn-m001:**

1. Navigate to the directory with the image downloaded in the previous section. 

1. Start up a webserver.
   
   ```
   ncn-m001# python3 -m http.server 8770
   ```

1. Update the BMC.
   
   ```
   m001# curl -k -u root:passwd https://ipaddressOfBMC/redfish/v1/UpdateService/Actions/SimpleUpdate -d ‘{“ImageURI”:”http://ipaddressOfM001:8770/filename”, ”TransferProtocol”:”HTTP”, ”UpdateComponent”:”BMC”}’
   ```
   
   The following information is used in the command:

   * `passwd` = Root password of the BMC
   * `ipaddressOfBMC` = IP address of the BMC
   * `ipaddressOfM001` = IP address of ncn-m001
   * `filename` = Filename of the downloaded image

1. Update the BIOS.
   
   ```
   ncn-m001# curl -k -u root:passwd https://ipaddressOfBMC/redfish/v1/UpdateService/Actions/SimpleUpdate -d ‘{“ImageURI”:”http://ipaddressOfM001:8770/filename”, ”TransferProtocol”:”HTTP”, ”UpdateComponent”:”BIOS”}’
   ```

   The following information is used in the command:

   * `passwd` = Root password of the BMC
   * `ipaddressOfBMC` = IP address of the BMC
   * `ipaddressOfM001` = IP address of ncn-m001
   * `filename` = Filename of the downloaded image

1. Reboot ncn-m001 after updating the BIOS. Follow the instructions in the [Reboot NCNs](../node_management/Reboot_NCNs.md) for rebooting ncn-m001.